src/compute_jacobian_ad.jl

```diff
@@ -8,20 +8,47 @@ struct ForwardColorJacCache{T,T2,T3,T4,T5}
     color::T5
 end
 
-function ForwardColorJacCache(f,x;
+function default_chunk_size(maxcolor)
+    if maxcolor < DEFAULT_CHUNK_THRESHOLD
+        Val(maxcolor)
+    else
+        Val(DEFAULT_CHUNK_THRESHOLD)
+    end
+end
+
+getsize(::Val{N}) where N = N
+getsize(N::Integer) = N
+
+function ForwardColorJacCache(f,x,_chunksize = nothing;
                               dx = nothing,
                               color=1:length(x))
 
-    t = zeros(Dual{typeof(f), eltype(x), maximum(color)},length(x))
+    if _chunksize === nothing
+        chunksize = default_chunk_size(maximum(color))
+    else
+        chunksize = _chunksize
+    end
+
+    t = zeros(Dual{typeof(f), eltype(x), getsize(chunksize)},length(x))
 
     if dx === nothing
         fx = similar(t)
         _dx = similar(x)
     else
-        fx = zeros(Dual{typeof(f), eltype(dx), maximum(color)},length(dx))
+        fx = zeros(Dual{typeof(f), eltype(dx), getsize(chunksize)},length(dx))
         _dx = dx
     end
 
+    p = generate_chunked_partials(x,color,chunksize)
+    ForwardColorJacCache(t,fx,_dx,p,color)
+end
+
+generate_chunked_partials(x,color,N::Integer) = generate_chunked_partials(x,color,Val(N))
+function generate_chunked_partials(x,color,::Val{N}) where N
+
+    # TODO: should only go up to the chunksize each time, and should
+    # generate p[i] different parts, each with less than the chunksize
+
     partials_array = Array{Bool}(undef, length(x), maximum(color))
```

Comment:
Could this be lighter on memory:
```suggestion
    partials_array = BitMatrix{Bool}(undef, length(x), maximum(color))
```